editors/code/src/utils/diagnostics/SuggestedFixCollection.ts

```diff
@@ -0,0 +1,74 @@
+import * as vscode from 'vscode';
+import SuggestedFix from './SuggestedFix';
+
+/**
+ * Collection of suggested fixes across multiple documents
+ *
+ * This stores `SuggestedFix` model objects and returns them via the
+ * `vscode.CodeActionProvider` interface.
+ */
+export default class SuggestedFixCollection
+    implements vscode.CodeActionProvider {
+    public static PROVIDED_CODE_ACTION_KINDS = [vscode.CodeActionKind.QuickFix];
+
+    private suggestedFixes: Map<string, SuggestedFix[]>;
```

Comment:
Let's add a comment explaining that `string` here is an URL? I wish one day we could just put an object into a Map in JavaScript